Tests/Integration/MailosaurClient.swift

```diff
@@ -70,35 +70,26 @@ internal class MailosaurAPIClient {
         return "Basic: \(apiKey!)"
     }
     
-    func getMostRecentMagicLink() async throws -> String {
-        do{
-            let messages = try await listMessages()
-            guard !messages.isEmpty else { return "" }
-            let message = try await getMessage(id: messages[0].id)
-            guard !message.html.links.isEmpty else { return "" }
-            let incomingURL = message.html.links[0].href
-            let components = NSURLComponents(url: URL(string: incomingURL)!, resolvingAgainstBaseURL: true)
-            guard let magicLink = components!.queryItems?.filter({$0.name == "psg_magic_link"}).first?.value else {
-                return ""
-            }
-            return magicLink
-        } catch {
-            print(error)
-            return ""
-        }
+    func getMostRecentMagicLink() async -> String? {
+        guard let messages = try? await listMessages(),
+              !messages.isEmpty,
+              let message = try? await getMessage(id: messages[0].id),
+              !message.html.links.isEmpty,
+              let incomingURL = URL(string: message.html.links[0].href),
```

Comment:
Is there any way to get around hard-coding the index? Not a big deal but if the layout changes could tests break?

Reply:
In this case we ALWAYS want the latest message, so it will always be zero :)